### DISCUSSION TITLE
How to import json files in Isar database.

### DISCUSSION THREAD

**Discussion by siddhesh-nalawade** [isar/isar#525] [Help]
I have some nested complex json files. I want to load these json files into isar database. Also I want to implement full text search feature for these json files. I tried to create model class for these jsons but jsons are really complex. How should I load these json files in Isar and implement full text search feature.

Structure of JSON:

```
{
"mainCategoryName": ,
"options":[
              {
              "name":
              "details":[
                            {
                            "title"
                            "detailsArray": [
                                        {
                                         "BackgroundColour": ,
                                          "name": ,
                                          "information": []
                                        }
                                  ]
                             }
                        ]
                  }
            ],
"definition": [],
"id":,
"symptomName":
}
```

**Comment by simc** [comment]
I can't give you a detailed solution here but you can either:
- use links for nested structures
- normalize/flatten the json
- wait for embedded object support
